.github/workflows/source_aws.yml

```diff
@@ -47,12 +47,26 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
-      - name: Set up Go 1.x
+      - name: Set up Go
+        id:   set-up-go
         uses: actions/setup-go@v3
         with:
           go-version-file: plugins/source/aws/go.mod
           cache: true
           cache-dependency-path: plugins/source/aws/go.sum
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        with:
+          path:         |
+                        ~/.cache/go-build
```

Comment:
Aren't the build artifacts already cached by `actions/setup-go`? See https://github.com/actions/setup-go/tree/41c2024c46acfe1d0b8c9b7f20e28406983e553b#caching-dependency-files-and-build-outputs and also https://github.com/actions/setup-go/blob/41c2024c46acfe1d0b8c9b7f20e28406983e553b/src/package-managers.ts#L13

Looks like it caches the result of `go env GOMODCACHE` (module cache) and `go env GOCACHE` (build cache).
`GOCACHE` seems to resolve to `~/.cache/go-build` in the GitHub action runner:
https://github.com/cloudquery/cloudquery/actions/runs/4607636489/jobs/8142407370#step:3:30
![image](https://user-images.githubusercontent.com/26760571/229834363-0039bb45-f35c-4df6-ba9d-f3f8aa150d04.png)

Reply:
Also 
![image](https://user-images.githubusercontent.com/26760571/229835505-98899b8e-ad01-450d-b36a-5549f98cadf5.png)

Reply:
I agree and I tried to remove the custom cache to use the setup-go cache now